lib/ansible/galaxy/dependency_resolution/dataclasses.py

```diff
@@ -397,7 +397,6 @@ def from_requirement_dict(cls, collection_req, art_mgr, validate_signature_optio
                 req_type = 'dir'
                 req_source = src_path
             elif _is_collection_namespace_dir(src_path):
-                req_name = None  # No name for a virtual req or "namespace."?
```

Comment:
This value influences some decisions in the logic below. I'm afraid it'll cause some unintended side effects in poorly-tested cases. I remember being frustrated when I was first working on this decision tree. I'd feel much better if we had 100% coverage on the logic in this classmethod... (everything prior to the `return` statement)

Is it possible to leave it out of this patch? Or maybe work on getting everything covered?

Reply:
I *think* this one is safe to remove, but I didn't remove an earlier place we set req_name to None in this method, because I thought it would cause issues too. It's a complicated method. I'd like to avoid introducing an inconsistent warning, so I will try to work on getting 100% coverage.

Reply:
Wonderful! That'd put my mind at ease..